File: employee-backend/src/main/java/com/employee/service/AdminService.java
package com.employee.service;

import java.util.List;

import com.employee.Dto.CommentDto;
import com.employee.Dto.EmployeeDto;
import com.employee.Dto.TaskDto;

public interface AdminService {
	
	List<EmployeeDto> getAllEmps();
	
	EmployeeDto getEmployeeById(long id);
	
	EmployeeDto updateEmployee(long id , EmployeeDto employee);
	
	void deleteEmployee(long id);
	
	List<EmployeeDto> searchEmployeeByName(String name);
	
	 EmployeeDto getEmpFromToken(String token);
	 
	 /* Task  */
	 
	  TaskDto  createTask(TaskDto taskDto);
	  
	 List<TaskDto> getAllTasks();
	 
	 TaskDto updateTask(long id , TaskDto taskDto );
	 
	 TaskDto getTaskById(long id);
	 
	 void delete(long id);
	 
	 TaskDto updateTask(long id , String status);
	 
	////////////////////// Comments ////////////////////////
	
	 CommentDto createComment(long taskId , String content,String token);
	 
	 List<CommentDto> getAllComments(long taskId);


}

File: frontEnd/Employee_System/src/app/modules/admin/services/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage.service';

//const BASE_URL ="http://localhost:8080/api/admin";

@Injectable({
  providedIn: 'root'
})
export class AdminService {

private baseUrl ="http://localhost:8080/api/admin"

  constructor(private httpClient : HttpClient) { }

  getAllEmployees():Observable<any>{

    return this.httpClient.get(`${this.baseUrl}/employees`,{
      headers: this.createHeaders()
      });
  }

  getEmployeeById(id:number):Observable<any>{
    return this.httpClient.get(`${this.baseUrl}/employees/${id}`, {
      headers: this.createHeaders()
    });
  }

  updateEmployee(id:number , employee :any):Observable<any>{
    return this.httpClient.put(`${this.baseUrl}/employees/${id}`, employee, {
      headers: this.createHeaders()
    });
  }

  deleteEmployee(id:number):Observable<any>{
    return this.httpClient.delete(`${this.baseUrl}/employees/${id}` ,{
      headers: this.createHeaders()
    })
  }

  /**************  Task ********************* */


  createTask(taskDto:any):Observable<any>{
    return this.httpClient.post(`${this.baseUrl}/task` , taskDto ,{
      headers:this.createHeaders()
    })
  }

  getTaskById(id:number):Observable<any>{
    return this.httpClient.get(`${this.baseUrl}/task/${id}`, {
      headers: this.createHeaders()
    });
  }


  getAllTasks():Observable<any>{
    return this.httpClient.get(`${this.baseUrl}/tasks` ,{
      headers:this.createHeaders()
    })
  }

  updateTask(id:number , taskDto :any):Observable<any>{
    return this.httpClient.put(`${this.baseUrl}/task/${id}` , taskDto ,{
      headers:this.createHeaders()
    })
  }

  deleteTask(id:number){
    return this.httpClient.delete(`${this.baseUrl}/task/${id}` ,{
      headers:this.createHeaders()
    })
  }

  //////////////////// Comments /////////////////

  createComment(id :number , content: string):Observable<any>{
    const params ={
      content : content
    }
  return this.httpClient.post(`${this.baseUrl}/task/comment/${id}` , null,{
    params : params ,
    headers: this.createHeaders()
  })
  }

  getAllCommentsOfTask(id:number):Observable<any>{
    return this.httpClient.get(`${this.baseUrl}/comments/${id}` ,{
      headers: this.createHeaders()
    })
  }

  private createHeaders():HttpHeaders{
    return new HttpHeaders().set(
      "Authorization" , "Bearer "+StorageService.getToken());
  }

  


    

}
    